vnext/Microsoft.ReactNative/ReactNativeAppBuilder.cpp

```diff
@@ -38,29 +43,69 @@ winrt::ReactNative::ReactNativeAppBuilder ReactNativeAppBuilder::SetReactInstanc
 
 winrt::ReactNative::ReactNativeAppBuilder ReactNativeAppBuilder::SetCompositor(
     winrt::Microsoft::UI::Composition::Compositor const &compositor) {
-  m_compositor = compositor;
-
+  m_reactNativeWin32App.as<implementation::ReactNativeWin32App>().get()->Compositor(compositor);
   return *this;
 }
 
 winrt::ReactNative::ReactNativeAppBuilder ReactNativeAppBuilder::SetAppWindow(
     winrt::Microsoft::UI::Windowing::AppWindow const &appWindow) {
-  m_appWindow = appWindow;
+  m_reactNativeWin32App.as<implementation::ReactNativeWin32App>().get()->AppWindow(appWindow);
+
+  return *this;
+}
+
+winrt::Microsoft::ReactNative::ReactNativeAppBuilder ReactNativeAppBuilder::SetReactViewOptions(
+    winrt::Microsoft::ReactNative::ReactViewOptions const &reactViewOptions) {
+  m_reactViewOptions = reactViewOptions;
 
   return *this;
 }
 
-winrt::ReactNative::ReactNativeHost ReactNativeAppBuilder::Start() {
-  auto hwnd{winrt::UI::GetWindowFromWindowId(m_appWindow.Id())};
+winrt::ReactNative::ReactNativeWin32App ReactNativeAppBuilder::Build() {
+  if (m_reactNativeWin32App.Compositor() == nullptr) {
+    m_reactNativeWin32App.as<implementation::ReactNativeWin32App>().get()->Compositor(nullptr);
```

Comment:
Not really a fan of the setting it to nullptr to tell it to create it's own. Just create it here.